### ISSUE TITLE
Relokasi Armada Antar Cabang (Linked Record & Partial Unique Index)

### ISSUE THREAD

**Issue by rifqiArdiansyah1** [rifqiArdiansyah1/rental-management-system#23]
**Tujuan:**
Pemisahan riwayat operasional, keuangan, dan booking masa lalu saat armada dipindahkan antar cabang tanpa memutus silsilah fisik kendaraan (*chain of custody*), serta pelepasan `@unique` global plat nomor menjadi *Partial Unique Index* di PostgreSQL.

### Background & Problem
1. **In-place update (`branchId = newBranch`)**: Mendistorsi laporan finansial dan riwayat booking cabang asal karena seluruh booking masa lalu ikut teratribusi ke cabang baru.
2. **Row baru terpisah (*Orphaned Record*)**: Memutus rekam jejak fisik mobil bagi auditor, dan terbentur constraint `@unique` plat nomor di PostgreSQL.
3. **Dual-path `moved`**: Mengeliminasi ketergantungan status `moved` pada `VehicleUnavailability` yang membingungkan secara semantik dan menyisakan celah "jendela tak terbatas".

### Proposed Changes

**1. Database Layer (`prisma/schema.prisma` & Migrations)**
- Model `Vehicle`:
  - Lepas `@unique` dari `plateNumber`, ganti dengan `@@index([plateNumber])`.
  - Tambahkan self-relation 1-ke-1: `previousVehicleId String? @unique` (`previousVehicle` & `relocatedTo`).
- Model `VehicleUnavailability`:
  - Hapus nilai `moved` dari enum `VehicleUnavailabilityReason` (fokus murni untuk `maintenance` / transit).
- Migrasi SQL `20260911000000_vehicle_relocation_chain/migration.sql`:
  - Drop constraint `Vehicle_plateNumber_key`.
  - Buat partial unique index:
    `CREATE UNIQUE INDEX "Vehicle_plateNumber_active_unique" ON "Vehicle" ("plateNumber") WHERE "isActive" = true;`
  - Preservasi data historis record existing bertipe `moved` (menambahkan `[Catatan migrasi: dulunya alasan moved]` ke `note` sebelum migrasi enum).
  - Update `run-migration.ts`.

**2. Backend Layer (`src/actions/adminVehicle.ts` & `src/lib/`)**
- Server Action atomik `relocateVehicle(sourceVehicleId, targetBranchId, options)` khusus `admin_pusat`:
  - Guard pesanan aktif (`ongoing`, `confirmed`, `pending_payment` dengan `endDate >= now`).
  - Deaktivasi unit asal (`isActive = false`, `status = 'moved'`), tutup unavailability lama dengan audit note.
  - Buat unit baru di cabang target dengan plat & spesifikasi sama (`isActive = true`, `status = 'available'`, `previousVehicleId = sourceVehicleId`).
  - Opsi transit window: jika ada durasi transit, unit baru otomatis diset `maintenance` dengan `estimatedEndAt = transitUntil`.
  - Audit log terstruktur `vehicle.relocated` (mencakup metadata penutupan unavailability lama).
- Penutupan total jalur lama: `updateVehicleStatus(id, 'moved')` ditolak keras.
- Helper terpusat `isPlateUniqueViolation(error)` menangani race condition constraint di `createVehicle`, `updateVehicle`, `softDeleteVehicle`, dan `relocateVehicle`.
- Bersihkan dead code `reason === 'moved'` di `src/lib/booking.ts`.
- Sesuaikan `prisma/seed.ts`, `tests/e2e/branch-scoping.spec.ts`, dan `scripts/normalize-vehicles.ts` dari `upsert`/`findUnique` berbasis plat nomor ke `findFirst`.

**3. UI Layer (`src/app/admin/(authenticated)/vehicles/`)**
- Modal Ubah Status: Hapus opsi `moved` (hanya tersisa `available` dan `maintenance`).
- Tambahkan aksi & modal khusus "Mutasi Antar Cabang" untuk `admin_pusat`.
- Render lineage badge silsilah pada tabel & kartu: `MUTASI ➔ [Cabang Baru]` dan `↳ Mutasi dari [Cabang Asal]`.

### Verification Plan
- [ ] Constraint partial unique index di Postgres menolak double active plate, tapi mengizinkan plate yang sama pada unit nonaktif.
- [ ] Dua `createVehicle` bersamaan dengan plat yang sama ditangani ramah lewat helper `isPlateUniqueViolation`.
- [ ] `updateVehicleStatus(id, 'moved')` langsung ditolak keras.
- [ ] Relokasi armada via `relocateVehicle` sukses membuat unit baru linked ke unit lama (`previousVehicleId`) dan menonaktifkan unit lama (`isActive: false`, `status: 'moved'`).
- [ ] Preservasi catatan unavailability lama jika unit sedang maintenance saat dimutasi.
- [ ] Guard booking aktif di masa depan menolak relokasi.
- [ ] Audit log `vehicle.relocated` mencatat lengkap `sourceVehicleId`, `newVehicleId`, cabang asal, cabang tujuan, dan riwayat penutupan unavailability.
- [ ] Seluruh test suite regresi lulus (`vehicle-status-consistency`, `branch-scoping`, `vehicle-maintenance-unavailability`).

**Comment by rifqiArdiansyah1** [comment]
﻿Implementasi dan verifikasi fitur Relokasi Armada Antar Cabang (Linked Record & Partial Unique Index) telah selesai dan dipush ke branch main via commit b3cd2fd.

### Ringkasan Eksekusi:
1. **Database & Schema**:
   - Model `Vehicle` kini memiliki self-relation 1-ke-1 (`previousVehicleId String? @unique`).
   - PostgreSQL Partial Unique Index (`Vehicle_plateNumber_active_unique`) aktif pada `plateNumber` dengan syarat `"isActive" = true`.
   - Enum `VehicleUnavailabilityReason` murni hanya untuk `maintenance` (transit window).
   - Seluruh data historis `moved` pada record sebelumnya dipreservasi dengan catatan audit.
2. **Backend & Domain Rules**:
   - Server Action atomik `relocateVehicle` mengunci unit dengan `SELECT FOR UPDATE`, menolak mutasi jika ada booking aktif masa depan, mendeaktivasi unit lama, menerbitkan unit baru terhubung, dan mencatat `vehicle.relocated` di audit log.
   - Penutupan total mutasi manual lewat status generik (`newStatus === 'moved'` ditolak).
   - Helper `isPlateUniqueViolation` menangani Prisma `P2002`, Postgres error `23505`, dan partial index violation.
3. **UI/UX**:
   - Modal Ubah Status dibersihkan dari opsi `moved`.
   - Tombol dan Modal khusus "Mutasi Cabang" disediakan khusus bagi `admin_pusat`.
   - Badge visual silsilah armada terpasang di tabel dan kartu (`MUTASI ➔ [Cabang Baru]` dan `↳ Mutasi dari [Cabang Asal]`).
4. **Hasil Verifikasi Otomatis (100% Green)**:
   - `tests/e2e/vehicle-relocation.spec.ts` (7/7 PASSED)
   - `tests/e2e/vehicle-status-consistency.spec.ts` (5/5 PASSED)
   - `tests/e2e/vehicle-maintenance-unavailability.spec.ts` (6/6 PASSED)
   - `tests/e2e/branch-scoping.spec.ts` (2/2 PASSED)
   - `npx tsc --noEmit` (0 Errors)
